nestor_api/lib/config.py

```diff
@@ -63,13 +61,23 @@ def _resolve_variable(template: str, variables: dict, path: str) -> str:
 
     # Resolve pattern '{{variable}}'
     referenced_names = re.findall(r"{{([\w\-.]+)}}", template)
-    for var_name in referenced_names:
-        var_value = variables.get(var_name)
-        if var_value is not None:
-            if not isinstance(var_value, str):
-                raise ConfigurationError(path, "Referenced variable should resolved to a string")
-
-            final_value = re.sub(f"{{{{{var_name}}}}}", var_value, final_value)
+    if len(referenced_names) > 0:
+        for var_name in referenced_names:
+            var_value = variables.get(var_name)
+            if var_value is not None:
+                if not isinstance(var_value, str):
+                    raise ConfigurationError(
+                        path, "Referenced variable should resolved to a string"
+                    )
+
+                final_value = re.sub(f"{{{{{var_name}}}}}", var_value, final_value)
+
+    # Resolve pattern '$ENV_VAR'
+    env_variables = re.findall(r"^\$(\w+)$", template)
+    if len(env_variables) > 0:
+        env_var = os.environ.get(env_variables[0])
```

Comment:
We will never have more than one en variable?

Reply:
The only use case I encountered for this is when I need to reference a build-time variable (a `NPM_TOKEN` in my case).
The regex will match the whole content, so "xxx $VAR xxx" won't be resolved, only 1 var supported.

I don't feel like allowing some patterns like `$SUB_DOMAIN.$DOMAIN" would be very nice but it is really just my opinion